mesheryctl/pkg/utils/helpers.go

```diff
@@ -591,3 +593,53 @@ func ValidateURL(URL string) error {
 	}
 	return nil
 }
+
+// ListManifests lists the manifest files stored in GitHub
+func ListManifests(url string) ([]Manifest, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to make GET request to %s", url)
+	}
+	defer SafeClose(resp.Body)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read response body")
+	}
+
+	var manLis ManifestList
+
+	json.Unmarshal([]byte(body), &manLis)
+	// log.Info(manLis.Tree[3].Path)
+	return manLis.Tree, nil
+}
+
+// GetManifestURL returns the URLs for the manifest files
+func GetManifestURL(manifest Manifest, rawManifestsURL string) string {
+	var manifestURL string
+
+	if manifest.Typ == "blob" {
+		manifestURL = rawManifestsURL + manifest.Path
+		return manifestURL
+	}
+	return ""
+}
+
+// DownloadManifests downloads all the Kubernetes manifest files
+func DownloadManifests(manifestArr []Manifest, rawManifestsURL string) error {
+	if _, err := os.Stat(MesheryFolder); os.IsNotExist(err) {
+		if err := os.Mkdir(MesheryFolder, 0777); err != nil {
+			return errors.Wrapf(err, SystemError(fmt.Sprintf("failed to make %s directory", MesheryFolder)))
+		}
+	}
+	for _, manifest := range manifestArr {
+		if manifestFile := GetManifestURL(manifest, rawManifestsURL); manifestFile != "" {
+			filepath := MesheryFolder + "/" + manifest.Path
+			log.Info(manifestFile, "\t", filepath, "\n")
+			if err := DownloadFile(filepath, manifestFile); err != nil {
+				return errors.Wrapf(err, SystemError(fmt.Sprintf("failed to download %s file from %s", filepath, manifestFile)))
+			}
+		}
+	}
+	return nil
+}
```

Comment:
So, I tried to create a `manifests` folder but failed and had to revert to just downloading all the files directly to `.meshery` folder. That works fine but when I try to create the `manifests` folder, it throws an error.

I tried by replacing `MesheryFolder` with `ManifestsFolder`. Any idea on how to fix this would be really great.

Reply:
What was the code, and what was the error?
Btw, concatenating paths like that will fail on Windows. path.Join should be used.

Reply:
@mgfeller Changed it to path.join(). I was able to fix the error.